### ISSUE TITLE
Linter seems to always think it's in the wrong directory (1 parent up)

### ISSUE THREAD

**Issue by dominicbarnes** [AtomLinter/linter-eslint#355]
Using `atom@1.2.4` and `linter-eslint@5.2.5`, I'm getting errors when using [eslint-plugin-require-path-exists](https://www.npmjs.com/package/eslint-plugin-require-path-exists). When linter-eslint is running, it always seems to think it's 1 directory higher than it really is, so all my relative imports show as being broken. For example:

<img width="831" alt="screen shot 2015-12-06 at 3 54 01 pm" src="https://cloud.githubusercontent.com/assets/160197/11616490/9da18554-9c31-11e5-94e2-cdd998386199.png">

It appears like the path it is checking is relative to `~/Github/makojs` instead of `~/Github/makojs/server`. (up 1 directory) I do not see this error when running `eslint` directly, so I believe this is an issue with `linter-eslint` somehow getting the `pwd` confused?

**Comment by IanVS** [comment]
This should have been fixed in https://github.com/AtomLinter/linter-eslint/pull/340.  Are you using the latest version of `linter-eslint`?

**Comment by Arcanemagus** [comment]
Can you try out the changes from #350 and see if things are resolved with that for you?

**Comment by dominicbarnes** [comment]
I'm not well-versed in atom dev, and I don't really have the time during the day today, I'm afraid.

**Comment by Arcanemagus** [comment]
Actually that shouldn't change this, nevermind on that. Are you using a `.eslintignore` file?

**Comment by dominicbarnes** [comment]
Aha! Something interesting, if there is a `.eslintignore`, everything works correctly. As soon as I delete it, everything starts throwing these errors.

**Comment by Arcanemagus** [comment]
Do you have a `.eslintignore` in `/Users/dominic/GitHub/makojs`? (I'm assuming while testing you placed one in `../lib/server`)

**Comment by dominicbarnes** [comment]
And another thing, if there is *not* a `.eslintignore` in the same directory (even if there is one at in a parent directory) it still fails. This means it works whenever each file has a sibling `.eslintignore`.

**Comment by dominicbarnes** [comment]
Oh, and I do *not* have a `$HOME/Github/makojs/.eslintignore`, or in any directory higher than that, I only have them once for each project.

**Comment by IanVS** [comment]
There can only be one `.eslintignore` file, at the root of your project.  This is a requirement for ESLint, their ignore files do not cascade.

**Comment by dominicbarnes** [comment]
@IanVS I'm doing exactly that, I only have 1 per project. After a bit more poking around, here is a complete description of my problem. Given the directory structure:

 - .eslintignore
 - index.js (imports ./lib/index.js)
 - lib/index.js (imports ./foo.js)
 - lib/foo.js

Here are the things that succeed/fail:

 - Lint `index.js`: success
 - Lint `lib/index.js`: **fail**
 - After deleting `.eslintignore`
    - Lint: `index.js`: **fail**

So, evidently the problem is related to `cwd` being set based on the `eslintignore` in some way. (sorry, I don't know the internals here)

**Comment by Arcanemagus** [comment]
Is your screenshot from `index.js`? If so it _should_ fail as you are attempting to import `../lib/server` instead of `./lib/server`.

**Comment by dominicbarnes** [comment]
Sorry, my screenshot wasn't necessarily from the example I just gave. (although it is similar) In my screenshot, I'm linting `bin/mako-serve` (but I get the same result as linting `lib/index.js`)

**Comment by Arcanemagus** [comment]
So if I understand you correctly you have this structure:
```
project
├── index.js (importing ./lib/libindex.js)
├── lib
│   ├── libindex.js (importing foo.js)
│   ├── foo.js
```
And when you open `libindex.js` it is trying to grab `foo.js` from the main project folder?

**Comment by dominicbarnes** [comment]
@Arcanemagus it sounds like you are trying to figure out if my relative paths are wrong, which would mean it *should* fail. As I already stated, I do **not** get the same error running ESLint manually, as my paths are not incorrect.

**Comment by Arcanemagus** [comment]
No, I'm trying to figure out how you have this set up to try to reproduce it so it can be debugged and fixed :wink:.

**Comment by dominicbarnes** [comment]
If it would be helpful, I can create a gist that would be easier than trying to copy/paste one of these comments.

**Comment by Arcanemagus** [comment]
Sure, that would be great.

**Comment by dominicbarnes** [comment]
[linter-eslint-bug-repro.zip](https://github.com/AtomLinter/linter-eslint/files/54781/linter-eslint-bug-repro.zip)

Extract that archive, run `npm install` and open up Atom in the directory. Navigate around, and you'll see which things fail or not. (I've added comments)

**Comment by Arcanemagus** [comment]
Awesome, thanks! Will take a look later tonight.

**Comment by Arcanemagus** [comment]
Sorry I didn't get a chance to track this down last night, looks like you found the root issue though.

**Comment by Arcanemagus** [comment]
This should be fixed with v1.0.25 of `eslint-plugin-require-path-exists` per the issue linked above, marking as closed.
